File: src/main/java/a/Test1.java
package a;

/**
 * @Author lzc
 * @Date 2020/4/20 16:57
 */
public class Test1 {
    public static void main(String[] args) {
        /*String a = "a";
        String m = a + "b";
        String n = a + "b";
        System.out.println(m == n);
        System.out.println(System.identityHashCode(m));
        System.out.println(System.identityHashCode(n));
        System.out.println(m.equals(n));*/
        int a = 10;
        int b = 20;
        // 其实是输出的 a = b 这个语句的值
        // 在java中, 赋值语句的值, 是赋值号, 左边的变量的值
        System.out.println(a = b);  // true 20

        foo2(1);
        foo2(1, 2);
        foo2(1, 2, 4);
        foo2(new int[]{1,2,3});

    }

    public void foo(int a){
        a = 100;
    }

    public void foo1(final int a){

    }

    public static void foo2(int ... ss){
        // ss 当做一个数组
    }

}

File: src/main/scala/com/atguigu/scala1128/day04/Test1.scala
package com.atguigu.scala1128.day04

/**
 * Author atguigu
 * Date 2020/4/24 8:55
 */
object Test1 {
def main(args: Array[String]): Unit = {
    // 1. 调用时
    foo1{
        println(" => Int  .....")
        1
    }
  val f =   foo2(() => {
        println(" () => Int")
        1
    })
    f()
}

def foo1(f:  => Int) = {
   val i =  f  // 表示执行传过来的代码块
    println(i)
    f
}

def foo2(f:() => Int) = {
    f()
    val a = 10
    () => println(a)
}
}
